libmscore/score.cpp

```diff
@@ -392,24 +392,25 @@ Score::~Score()
 
 Score* Score::clone()
 {
-    QBuffer buffer;
-    buffer.open(QIODevice::WriteOnly);
-    XmlWriter xml(this, &buffer);
-    xml.header();
+    Excerpt excerpt(masterScore());
+    excerpt.setParts(_parts);
+    excerpt.setTitle(title());
 
-    xml.stag("museScore version=\"" MSC_VERSION "\"");
-    write(xml, false);
-    xml.etag();
-
-    buffer.close();
+    for (Part* part : _parts) {
+        for (int i = part->startTrack(), j = 0; i < part->endTrack(); i++, j++) {
+            excerpt.tracks().insert(i, j);
+        }
+    }
 
-    XmlReader r(buffer.buffer());
-    MasterScore* score = new MasterScore(style());
-    score->read1(r, true);
+    Score* copy = new Score(masterScore());
+    excerpt.setPartScore(copy);
+    copy->_style = _style;
+    masterScore()->addExcerpt(&excerpt);
+    Excerpt::createExcerpt(&excerpt);
+    masterScore()->removeExcerpt(&excerpt);
+    excerpt.setPartScore(nullptr);
 
-    score->addLayoutFlags(LayoutFlag::FIX_PITCH_VELO);
-    score->doLayout();
```

Comment:
I'm wondering if whether we still need this

Reply:
They are called from Excerpt::createExcerpt